__tests__/app.test.js

```diff
@@ -1,44 +1,80 @@
+const request = require("supertest");
+const seed = require("../db/seeds/seed");
+const app = require("../app");
+const db = require("../db/connection");
+const data = require("../db/data/test-data/index");
+const endpoints = require("../endpoints.json");
 
-beforeAll(() => seed(data))
-afterAll(() => db.end())
+beforeAll(() => seed(data));
+afterAll(() => db.end());
 
-describe('GET /api - controller set up correctly', () => {
-    test('200: return an object which describes all available endpoints on the API', () => {
-        return request(app)
-        .get('/api')
-        .expect(200)
-        .then(({body}) => {
-            const responseEndpoints = body.endpoints
-            expect(responseEndpoints).toEqual(endpoints)
-        })
-    });
-    test('404: when invalid path given, returns error', () => {
-        return request(app)
-        .get('/api/notavalidpath')
-        .expect(404)
-        .then(({body}) => {
-            expect(body.msg).toBe('Not Found')
-        })
-    })
+describe("GET /api - controller set up correctly", () => {
+  test("200: return an object which describes all available endpoints on the API", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        const responseEndpoints = body.endpoints;
+        expect(responseEndpoints).toEqual(endpoints);
+      });
+  });
+  test("404: when invalid path given, returns error", () => {
+    return request(app)
+      .get("/api/notavalidpath")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
 });
 
-describe('GET /api/topics', () => {
-    test('200: return all topics from table, respond with an array of objects, each with a slug and description', () => {
-        return request(app)
-        .get('/api/topics')
-        .expect(200)
-        .then(({body}) => {
-            expect(body.topics).toHaveLength(3)
-            body.topics.forEach((topic) => {
-                expect(typeof topic.slug).toBe('string')
-                expect(typeof topic.description).toBe('string')
-            })
-        })
-    });
-});
\ No newline at end of file
+describe("GET /api/topics", () => {
+  test("200: return all topics from table, respond with an array of objects, each with a slug and description", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.topics).toHaveLength(3);
+        body.topics.forEach((topic) => {
+          expect(typeof topic.slug).toBe("string");
+          expect(typeof topic.description).toBe("string");
+        });
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("200: return articles by their id's", () => {
+    return request(app)
+      .get("/api/articles/3")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body;
+        expect(article).toHaveLength(1);
```

Comment:
You could use a flexible jest matcher like .toMatchObject() in this test which would allow to extension of the endpoint at a later day and make your test more flexible.